core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchWriter.java

```diff
@@ -753,15 +721,16 @@ private void addMutations(MutationSet mutationsToSend) {
         log.trace(String.format("Started sending %,d mutations to %,d tablet servers", count,
             binnedMutations.keySet().size()));
 
-      // randomize order of servers
-      ArrayList<String> servers = new ArrayList<>(binnedMutations.keySet());
-      Collections.shuffle(servers);
+      // order servers by decreasing number of mutations
+      TreeMap<Integer,String> servers = new TreeMap<>();
+      binnedMutations.forEach((server, tsm) -> servers.put(tsm.getMutations().size(), server));
```

Comment:
The more I think about this I am not sure if its a good ideal.  Its probably a good thing for a single batch writer writing to many tservers.  However for many batch writers it seems like it could lead to a degenerate case where all the batch writer write to a few tservers, where each batch writer randomly ordering tservers would avoid this.

Reply:
I reverted this change.